jx crashes after entering github username

no idea if this is reproducible or not but after I entered my github username (as part of the `jx create cluster gke` flow) jx crashed

```
nginx ingress controller installed and configured
Lets set up a git username and API token to be able to perform CI/CD

? GitHub username for CI/CD pipelines:
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x28 pc=0x166ec20]

goroutine 1 [running]:
github.com/jenkins-x/jx/pkg/auth.(*UserAuth).IsInvalid(...)
        /home/jenkins/go/src/github.com/jenkins-x/jx/pkg/auth/user_auth.go:26
github.com/jenkins-x/jx/pkg/jx/cmd.(*InstallOptions).getGitUser(0xc0420a8b28, 0x0, 0x0, 0x0, 0x0, 0x0)
        /home/jenkins/go/src/github.com/jenkins-x/jx/pkg/jx/cmd/install.go:708 +0x200
github.com/jenkins-x/jx/pkg/jx/cmd.(*InstallOptions).getGitToken(0xc0420a8b28, 0x7, 0x1b081dd, 0x6, 0x1b07a49, 0x5, 0x1b20c88)
        /home/jenkins/go/src/github.com/jenkins-x/jx/pkg/jx/cmd/install.go:627 +0xef
github.com/jenkins-x/jx/pkg/jx/cmd.(*InstallOptions).getGitSecrets(0xc0420a8b28, 0x0, 0x0, 0x1b0a6c0, 0x7)
        /home/jenkins/go/src/github.com/jenkins-x/jx/pkg/jx/cmd/install.go:585 +0x4a
github.com/jenkins-x/jx/pkg/jx/cmd.(*InstallOptions).Run(0xc0420a8b28, 0xc042072960, 0x22)
        /home/jenkins/go/src/github.com/jenkins-x/jx/pkg/jx/cmd/install.go:292 +0x72e
github.com/jenkins-x/jx/pkg/jx/cmd.(*CreateClusterOptions).initAndInstall(0xc0420a8a80, 0x1b060e0, 0x3, 0xc042072960, 0x10)
        /home/jenkins/go/src/github.com/jenkins-x/jx/pkg/jx/cmd/create_cluster.go:150 +0x6f
github.com/jenkins-x/jx/pkg/jx/cmd.(*CreateClusterGKEOptions).createClusterGKE(0xc0420a8a80, 0x1b060e0, 0x3)
        /home/jenkins/go/src/github.com/jenkins-x/jx/pkg/jx/cmd/create_cluster_gke.go:258 +0x6a9
github.com/jenkins-x/jx/pkg/jx/cmd.(*CreateClusterGKEOptions).Run(0xc0420a8a80, 0xc0426e1c50, 0x0)
        /home/jenkins/go/src/github.com/jenkins-x/jx/pkg/jx/cmd/create_cluster_gke.go:118 +0x6e
github.com/jenkins-x/jx/pkg/jx/cmd.NewCmdCreateClusterGKE.func1(0xc042136c80, 0x28899b0, 0x0, 0x0)
        /home/jenkins/go/src/github.com/jenkins-x/jx/pkg/jx/cmd/create_cluster_gke.go:89 +0x7a
github.com/jenkins-x/jx/vendor/github.com/spf13/cobra.(*Command).execute(0xc042136c80, 0x28899b0, 0x0, 0x0, 0xc042136c80, 0x28899b0)
        /home/jenkins/go/src/github.com/jenkins-x/jx/vendor/github.com/spf13/cobra/command.go:766 +0x2c8
github.com/jenkins-x/jx/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc04209b180, 0xc042281180, 0xc042036f00, 0xc04212e500)
        /home/jenkins/go/src/github.com/jenkins-x/jx/vendor/github.com/spf13/cobra/command.go:852 +0x33b
github.com/jenkins-x/jx/vendor/github.com/spf13/cobra.(*Command).Execute(0xc04209b180, 0xc0426ed940, 0x272d9a0)
        /home/jenkins/go/src/github.com/jenkins-x/jx/vendor/github.com/spf13/cobra/command.go:800 +0x32
github.com/jenkins-x/jx/cmd/jx/app.Run(0xc042022000, 0x16d1190)
        /home/jenkins/go/src/github.com/jenkins-x/jx/cmd/jx/app/jx.go:18 +0x97
main.main()
        /home/jenkins/go/src/github.com/jenkins-x/jx/cmd/jx/jx.go:10 +0x29

```

--- a/pkg/auth/config.go
+++ b/pkg/auth/config.go
@@ -244,11 +244,15 @@ func (c *AuthConfig) PickServerUserAuth(server *AuthServer, message string, batc
 			Message: message,
 			Options: usernames,
 		}
-		err := survey.AskOne(prompt, &username, nil)
+		err := survey.AskOne(prompt, &username, survey.Required)
 		if err != nil {
 			return &UserAuth{}, err
 		}
-		return m[username], nil
+		answer := m[username]
+		if answer == nil {
+			return nil, fmt.Errorf("No username chosen!")
+		}
+		return answer, nil
 	}
 	return &UserAuth{}, nil
 }

